### ISSUE TITLE
Replicated external choice should be allowed to have zero or one enumeration values

### ISSUE THREAD

**Issue by joey-coleman** [symphonytool/symphony#155]
> The following bug was [originally reported](https://sourceforge.net/p/compassresearch/bugs/155/) on Sourceforge by [*anonymous](https://sourceforge.net/u/*anonymous/profile/), 2013-11-06 10:17:22.572000:

I believe the process below should offer only test.1, then stop. In 
other words, a replicated external choice restricted to one process just offers 
that process.

```
process Hobson =
  begin

  actions
   Test = [] i in set {1} @ test.i -> Stop
   @
   Test
end
```

Also, I would expect the process below to be equivalent to Stop. In 
other words, a replicated choice restricted to no processes offers no 
processes.

```
process RepChoice2 =
  begin

  actions
   Test = [] i in set {} @ test.i -> Stop
   @
   Test
end
```

**Comment by joey-coleman** [comment]
> [Comment](https://sourceforge.net/p/compassresearch/bugs/155/#0137) by [iwdy23](https://sourceforge.net/u/iwdy23/profile/), 2013-11-06 10:18:08.275000:

I will be out of the office until the 9th of November and don't know the quality of my internet connection till then so I may not be able to read and reply emails until then in a timely manner.

Best wishes,
Peter

**Comment by joey-coleman** [comment]
> [Comment](https://sourceforge.net/p/compassresearch/bugs/155/#54e7) by [joeycoleman](https://sourceforge.net/u/joeycoleman/profile/), 2013-11-06 10:23:48.060000:
- Description has changed:

Diff:

```

--- old
+++ new
@@ -1,9 +1,8 @@
-
 I believe the process below should offer only test.1, then stop. In 
 other words, a replicated external choice restricted to one process just offers 
 that process.

---
+~~~~
 process Hobson =
   begin

@@ -12,11 +11,13 @@
    @
    Test
 end
+~~~~

 Also, I would expect the process below to be equivalent to Stop. In 
 other words, a replicated choice restricted to no processes offers no 
 processes.

+~~~~
 process RepChoice2 =
   begin

@@ -25,3 +26,4 @@
    @
    Test
 end
+~~~~

```

**Comment by joey-coleman** [comment]
> [Comment](https://sourceforge.net/p/compassresearch/bugs/155/#0dbf) by [joeycoleman](https://sourceforge.net/u/joeycoleman/profile/), 2013-11-06 10:23:57.497000:
- **assigned_to**: Anders Kaels Malmos

**Comment by joey-coleman** [comment]
> [Comment](https://sourceforge.net/p/compassresearch/bugs/155/#fbe4) by [anderskm](https://sourceforge.net/u/anderskm/profile/), 2013-11-15 14:06:58.048000:
- **status**: open --> closed
- **Module**: unknown --> interpreter

**Comment by joey-coleman** [comment]
> [Comment](https://sourceforge.net/p/compassresearch/bugs/155/#eff3) by [anderskm](https://sourceforge.net/u/anderskm/profile/), 2013-11-15 14:06:58.559000:

this is now fixed in commmit: 774c385e235a1bdbe659bf452d2ebac6c6834071 and b033be7f1bf16841c900f464d5a941d3df19253e
